packages/next/src/lib/find-root.ts

```diff
@@ -18,5 +19,29 @@ export function findRootLockFile(cwd: string) {
 
 export function findRootDir(cwd: string) {
   const lockFile = findRootLockFile(cwd)
-  return lockFile ? dirname(lockFile) : undefined
+  if (!lockFile) return undefined
+
+  const lockFiles = [lockFile]
+  while (true) {
+    const nextDir = dirname(dirname(lockFiles[lockFiles.length - 1]))
+    const newLockFile = findRootLockFile(nextDir)
+
+    if (newLockFile) {
+      lockFiles.push(newLockFile)
+    } else {
+      break
+    }
+  }
+
+  // Only warn if not in a build worker to avoid duplicate warnings
+  if (typeof process.send !== 'function' && lockFiles.length > 1) {
+    Log.warnOnce(
+      `Warning: Found multiple lockfiles. Consider removing the lockfiles at ${lockFiles
+        .slice(0, lockFiles.length - 1)
+        .map((str) => '\n   * ' + str)
+        .join('')}\n`
+    )
```

Comment:
nit: maybe include the selected file in the message?
```suggestion
    Log.warnOnce(
      `Warning: Found multiple lockfiles. Selecting ${lockFile}. Consider removing the lockfiles at: ${lockFiles
        .slice(0, lockFiles.length - 1)
        .map((str) => '\n   * ' + str)
        .join('')}\n`
    )
```